dvc/rwlock.py

```diff
@@ -4,15 +4,20 @@
 from collections import defaultdict
 from contextlib import contextmanager
 
-from voluptuous import Schema, Invalid
+from voluptuous import Schema, Invalid, Required
 
 from .exceptions import DvcException
 from .lock import LockError
 from .utils import relpath
 
-INFO_SCHEMA = {"pid": int, "cmd": str}
+INFO_SCHEMA = {Required("pid"): int, Required("cmd"): str}
 
-SCHEMA = Schema({"write": {str: INFO_SCHEMA}, "read": {str: [INFO_SCHEMA]}})
+SCHEMA = Schema(
+    {
+        Required("write", default={}): {str: INFO_SCHEMA},
+        Required("read", default={}): {str: [INFO_SCHEMA]},
```

Comment:
So these used to be `Optional`(default) but now they are `Required`. Is this intended?

Sure, de-facto we are always setting even empty dicts for these down below, so it won't break anything in dvc itself. But on the other hand, these are not required by our reader and it is not required in terms of format, so I don't really see a point in making them `Required` now. The problem you've described in the PR description is fully fixed by `INFO_SCHEMA` adjustment you've done above. If you want to get rid of `defaultdict` then let's `Optional(..., default={})` here instead.

Reply:
@efiop , I think `Required` and `Optional` are the same when a `default` is provided, is just semantically different.

I'll make the change, since it makes more sense in that regard.